File: benchmarks/python/record_array_walk.py
from dataclasses import dataclass


@dataclass(slots=True)
class Point:
    x: int
    y: int


def main() -> None:
    points: list[Point] = []
    for i in range(5000):
        points.append(Point(i + 1, (i * 3) % 31))

    total = 0
    for point in points:
        total += point.x + point.y * 2

    print(total)


if __name__ == "__main__":
    main()

File: benchmarks/c/record_array_walk.c
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int64_t x;
    int64_t y;
} Point;

int main(void) {
    Point *points = malloc(sizeof(Point) * 5000);
    int64_t total = 0;

    if (points == NULL) {
        return 1;
    }

    for (int i = 0; i < 5000; ++i) {
        points[i].x = i + 1;
        points[i].y = (i * 3) % 31;
    }

    for (int i = 0; i < 5000; ++i) {
        total += points[i].x + points[i].y * 2;
    }

    free(points);
    printf("%lld\n", (long long) total);
    return 0;
}
